docs/proposal-cylc-clean.md

```diff
@@ -0,0 +1,251 @@
+# Cylc Clean
+
+Work on the new `cylc clean` is already underway.
+
+This command is able to locate workflow installations on different platforms
+and delete them.
+
+Ideally we would want `cylc clean` to be able to do a lot more.
+
+One, simple example is re-running workflows, which requires moving/removing the
+log dir and database. As per the
+[rose suite-run migration proposal](https://cylc.github.io/cylc-admin/proposal-rose-suite-run.html#cli-changes)
+this shall be done by cylc clean.
+
+> you will be able to use cylc clean to clean out the log directory
+> (and whatever other directories you want removing)
+
+This means that we need an interface in `cylc clean` to perform this.
+
+We could give users a pre-defined option, however, the exact requirements for
+this niche usage are hard to predict.
+
+This proposal suggests developing more generic functionality in `cylc clean`
+for this and other purposes. It also highlights some viable possibilities for
+future functionality which could see it largely surpass `rose_prune` and
+`rose_arch`. Not of top importance now but worth considering for interface
+design reasons.
+
+related https://github.com/cylc/cylc-flow/issues/3887
+closes https://github.com/cylc/cylc-flow/issues/1159
+
+(For reference `rose_arch` is ~500 lines of Python and does some things we
+might not want to do in Cylc).
+
+## Examples
+
+Control over which installations of a flow are cleaned:
+
+```
+# remove a workflow installation on the scheduler host
+cylc clean myflow --local
+
+# remove a workflow installation on any remote task hosts
+cylc clean myflow --remote
+
+# remove a workflow installation on the scheduler host and
+# on any remote task hosts
+cylc clean myflow
```

Comment:
Just to note that, as discussed in cylc/cylc-flow#3887, we'll need a `--force` option to handle the case where the remote clean cannot be completed.